### PR TITLE
feat: adjust github environment tracking to Segment

### PR DESCRIPTION
**Pull request by antelman107**

### CODE REVIEW

File: .github/workflows/report-workflow-stats.yml
Comment (duskpoet) [inline .github/workflows/report-workflow-stats.yml:1]: how is this change related?

**Review by duskpoet — APPROVED** [review]

File: .github/workflows/report-workflow-stats.yml
Comment (antelman107) [reply .github/workflows/report-workflow-stats.yml:1]: old commits from main. I don't understand myself. But I merged the real main and I'm going to squash this PR.
